### PR TITLE
Footnotes: autosave is not slashing JSON

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Fixes #53660, https://core.trac.wordpress.org/ticket/59103

## Why?

We're making this mistake:
https://developer.wordpress.org/reference/functions/update_post_meta/#character-escaping

This is what the REST API does:

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L393

### CODE REVIEW

File: packages/block-library/src/footnotes/index.php [hunk 1]
```diff
@@ -243,7 +243,7 @@ function _wp_rest_api_autosave_meta( $autosave ) {
 		return;
 	}
 
-	update_post_meta( $id, 'footnotes', $body['meta']['footnotes'] );
+	update_post_meta( $id, 'footnotes', wp_slash( $body['meta']['footnotes'] ) );
```
Comment (ramonjd) [inline packages/block-library/src/footnotes/index.php:246]: Auto saving still triggers the `count()` error for me, but not when I add `wp_slash()` to the update in `wp_save_footnotes_meta`, e.g., 


```php
update_metadata( 'post', $revision_id, 'footnotes', wp_slash( $footnotes ) );
```

I guess that's being called as well?

**Comment by ramonjd** [comment]
This addresses escaping new footnotes for me.

Here's what I'm seeing:

### Editor

Screenshot 2023-08-16 at 10 48 30 am

### Frontend
Screenshot 2023-08-16 at 10 48 34 am

### wp_postmeta
Screenshot 2023-08-16 at 10 49 05 am

✅ Save and autosave work and quotes are escaped by `addslashes` via `wp_slash`
✅ Frontend matches editor
✅ HTML and other chars are escaped (not by this PR, but just confirming)
✅ Checked in post and site editors

**Review by ramonjd — APPROVED** [review]
🦶🏻 🎶

**Comment by andrewserong** [comment]
Thanks for fixing this!

**Pull request merged**